JustSaying/Messaging/MessageHandling/BlockingMessageLock.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Threading.Tasks;
+
+namespace JustSaying.Messaging.MessageHandling
+{
+    /// <summary>
+    /// Used to convert "IMessageLock " instances into IMessageLockAsync
+    /// So that the rest of the system only has to deal with IMessageLockAsync
+    /// </summary>
+    public class BlockingMessageLock : IMessageLockAsync
+    {
+        public BlockingMessageLock(IMessageLock inner)
+        {
+            if (inner == null)
+            {
+                throw new ArgumentNullException(nameof(inner));
+            }
+
+            Inner = inner;
+        }
+
+        public IMessageLock Inner { get; }
+
+        public Task ReleaseLockAsync(string key) => Task.Run(() => Inner.ReleaseLock(key));
```

Comment:
Could this please be changed to:
```csharp
public Task ReleaseLockAsync(string key)
{
    Inner.ReleaseLock(key);
    return Task.Completed;
}
```

Reply:
We can't using CompletedTask for .NET 4.5.1

Reply:
Could use `Task.FromResult()` ?